### PR TITLE
Add Welcome Emails and contact us email

### LINKED ISSUE #94: Add contact us/feedback page to be sent to email
email: your.wisewallet@gmail.com

### LINKED ISSUE #104: Add welcome emails
When a user creates an account with a valid email (no safety measures for this) they will be send a welcome email.
image

### PR DESCRIPTION
**Pull request by aappacode**
When a user signs up, they will receive a welcome email via EmailJS. Also, when they fill out our contact form, we will receive an email at your.wisewallet@gmail.com, with their contact email.

### CODE REVIEW

**Comment by ravenclaw1999** [comment]
I am getting this error:

Screenshot 2023-12-04 at 2 21 43 AM

**Comment by aappacode** [comment]
> I am getting this error:
> 
> Screenshot 2023-12-04 at 2 21 43 AM

you have to do `npm install` before npm run dev (new dependency)

**Comment by ravenclaw1999** [comment]
So right now, it should only be sending the initial welcome email and only in-app notifications?

**Comment by ravenclaw1999** [comment]
> So right now, it should only be sending the initial welcome email and only in-app notifications?

Siema: "this is just for the contact us email form (bottom of home page) and sends email when a user makes a new account"

**Comment by ravenclaw1999** [comment]
Awesome, the welcome email goes through! The only thing is that the 'Start Saving' button on the email doesn't seem to lead anywhere.

**Comment by ravenclaw1999** [comment]
Would it be a good idea to add the contact form / footer on other pages as well? Or should we just have it on the home page?

**Comment by aappacode** [comment]
> Would it be a good idea to add the contact form / footer on other pages as well? Or should we just have it on the home page?

I wasn't sure how to do this---it would be nice!

**Comment by aappacode** [comment]
> Awesome, the welcome email goes through! The only thing is that the 'Start Saving' button on the email doesn't seem to lead anywhere.

what email service did you use? it works on outlook and gmail but I'll try to fix this since it's really on the EmailJS template side

**Comment by ravenclaw1999** [comment]
> > Awesome, the welcome email goes through! The only thing is that the 'Start Saving' button on the email doesn't seem to lead anywhere.
> 
> what email service did you use? it works on outlook for me

Gmail!

**Comment by ravenclaw1999** [comment]
> > Would it be a good idea to add the contact form / footer on other pages as well? Or should we just have it on the home page?
> 
> I wasn't sure how to do this hehe but it would be nice

I think you would just add the component in the App.jsx file. Similar to how Navbar is added.

`<HomeFooter />`

**Comment by ravenclaw1999** [comment]
Also, did you receive my contact form message?

**Comment by ravenclaw1999** [comment]
A small thing and not sure if you want to keep it separate from the web responsive design issues, but could you add a short media query to make sure the form is responsive?

Screenshot 2023-12-04 at 2 48 22 AM

**Comment by aappacode** [comment]
> A small thing and not sure if you want to keep it separate from the web responsive design issues, but could you add a short media query to make sure the form is responsive?
> 
> Screenshot 2023-12-04 at 2 48 22 AM

Hmm looks like this on my end:
image

update: saw you were using the inspect, let me try and if not i wont add

**Comment by aappacode** [comment]
> > > Awesome, the welcome email goes through! The only thing is that the 'Start Saving' button on the email doesn't seem to lead anywhere.
> > 
> > 
> > what email service did you use? it works on outlook for me
> 
> Gmail!

this is probably just a gmail protection and we can ignore it for now, but will try again

**Comment by aappacode** [comment]
> Also, did you receive my contact form message?

Screenshot 2023-12-04 at 3 34 57 AM

yes, and i made a typo on template (updated it)

File: src/pages/Register.jsx [hunk 1]
```diff
@@ -19,13 +22,29 @@ const Register = () => {
...
+            };
+            // send welcome email trigger
+            e.preventDefault();
+            emailjs.send('service_vcya2vm', 'welcome_1ywtamz', templateParameters, 'xZBCwlq2NxOnENizZ') // 'YOUR_SERVICE_ID', 'YOUR_TEMPLATE_ID', form.current, 'YOUR_PUBLIC_KEY'
+                .then(() => {
+                    alert("Your account has been created!")
+                }, () => {
+                    alert("Your account could not be created.")
+                });
+
```
Comment (ravenclaw1999) [inline src/pages/Register.jsx:37]: This code is repeating in Profile.jsx as well. Is there a reason we need it in both files?

**Comment by ravenclaw1999** [comment]
Since review comment can't be seen:

This code is repeating in Profile.jsx as well. Is there a reason we need it in both files?
`// send welcome email
            const templateParameters = { // these parameters are outlined in the template
                recipient: email,
            };
            // send welcome email trigger
            e.preventDefault();`

**Comment by ravenclaw1999** [comment]
Siema: One is for welcome email and one is for contact form.

**Review by ravenclaw1999 — APPROVED** [review]
Great, both the welcome email and contact form message goes through!

File: src/pages/Register.jsx (hunk 1 above)
Comment (aappacode) [reply src/pages/Register.jsx:37]: two different emails !

**Comment by aappacode** [comment]
Note: footer did not leverage the component as the footer had to look a bit different on each page, so the entire block of code was pulled over.

**Pull request merged**